index.js

```diff
@@ -79,6 +79,46 @@
 
   var _ = R.__;
 
+  /* istanbul ignore next */
+  var Any = function Any() {};
+
+  var curry = function(name, typePairs, f) {
+    return R.arity(typePairs.length, function() {
+      var args = arguments;
+      var $typePairs = [];
+      for (var idx = 0; idx < args.length; idx += 1) {
+        var typePair = typePairs[idx];
+        if (args[idx] === _) {
+          $typePairs.push(typePair);
+        } else if (typePair[1] !== Any && !R.is(typePair[1], args[idx])) {
+          throw new TypeError(R.join(' ', [
+            name,
+            'requires a value of type',
+            /^function (\w*)/.exec(typePair[1])[1],
+            'as its',
+            ['first', 'second', 'third', 'fourth', 'fifth'][typePair[0]],
```

Comment:
you really dont like functions that take many arguments ah... what happens if you're out of bound here?